src/runner.rs

```diff
@@ -139,9 +148,65 @@ fn parse_test(test_path: &Path, config: &TestConfig) -> InnerTestResult<Test> {
         expected_stdout,
         expected_stderr,
         expected_exit_status,
+        rest,
     })
 }
 
+fn overwrite_test(test_path: &PathBuf, config: &TestConfig, output: &Output, test: &Test) -> std::io::Result<()> {
+    // Maybe copy the file so we don't remove it if we fail here?
+    let mut file = File::create(test_path)?;
+
+    file.write_all(test.rest.trim_end().as_bytes())?;
+    writeln!(file, "")?;
+    writeln!(file, "")?;
+
+    if !test.command_line_args.is_empty() {
+        writeln!(file, "{}{}", config.test_args_prefix, test.command_line_args)?;
+        writeln!(file, "")?;
+    }
+
+    if Some(0) != output.status.code() {
+        writeln!(
+            file,
+            "{} {}",
+            config.test_exit_status_prefix,
+            output.status.code().unwrap_or(0)
+        )?;
+        writeln!(file, "")?;
+    }
+
+    // Doesn't handle \r correctly!
+    if output.stdout.len() != 0 {
+        writeln!(file, "{}", config.test_stdout_prefix)?;
+        for line in output.stdout.split(|c| *c == '\n' as u8) {
+            if line.is_empty() {
+                // Remove leading and trailing newlines
+                continue;
+            }
+            file.write_all(config.test_line_prefix.as_bytes())?;
+            file.write_all(line)?;
+            writeln!(file, "")?;
+        }
+        writeln!(file, "")?;
```

Comment:
Can you add a check to write all the output on the same line if it is only one line long to begin with? E.g. editing the all_keywords test then passing the overwrite flag will change it from

```py
# expected exit status: 3
# expected stdout: test

# expected stderr: error!
```
to
```py
# expected exit status: 3
# expected stdout:
# test

# expected stderr:
# error!
```

Reply:
Done.